neurom/view/_dendrogram.py

```diff
@@ -27,280 +27,165 @@
-    '''Vertices of a horizontal rectangle
+        Args:
+            neurom_section: NeuroM section tree.
+            dendrogram_root: root of dendrogram. This is a service arg, please don't set it on
+            your own. It is used to track cycles in ``neurom_section``.
+        '''
+        if dendrogram_root is None:
+            dendrogram_root = self
+            dendrogram_root.processed_section_ids = []
+        if neurom_section.id in dendrogram_root.processed_section_ids:
+            raise ValueError('Cycled morphology {}'.format(neurom_section))
+        dendrogram_root.processed_section_ids.append(neurom_section.id)
+
+        segments = neurom_section.points
+        segment_lengths = np.linalg.norm(
+            np.subtract(segments[:-1, COLS.XYZ], segments[1:, COLS.XYZ]), axis=1)
+        segment_radii = segments[:, COLS.R]
+
+        self.section_id = neurom_section.id
+        self.neurite_type = neurom_section.type
+        self.height = np.sum(segment_lengths)
+        self.width = 2 * np.max(segment_radii)
+        self.coords = Dendrogram.get_coords(segment_lengths, segment_radii)
+        self.children = [Dendrogram(child, dendrogram_root) for child in neurom_section.children]
+
+    @staticmethod
+    def get_coords(segment_lengths, segment_radii):
+        """Coordinates of dendrogram as polygon with respect to (0, 0) origin.
+
+        Args:
+            segment_lengths: lengths of dendrogram segments
+            segment_radii: radii of dendrogram segments
+
+        Returns:
+            (N,2) array of 2D x,y coordinates of Dendrogram polygon. N is the number of vertices.
+        """
+        y_coords = np.insert(segment_lengths, 0, 0)
+        y_coords = np.cumsum(y_coords)
+        x_left_coords = -segment_radii
+        x_right_coords = segment_radii
+        left_coords = np.hstack((x_left_coords[:, np.newaxis], y_coords[:, np.newaxis]))
+        right_coords = np.hstack((x_right_coords[:, np.newaxis], y_coords[:, np.newaxis]))
+        right_coords = np.flip(right_coords, axis=0)
+        return np.vstack((left_coords, right_coords))
+
+
+def create_dendrogram(neuron):
+    '''Creates a dendrogram for neuron
+
+    Args:
+        neuron (Neurite|Neuron): Can be a Neurite or a Neuron instance.
+
+    Returns:
+        Dendrogram of ``neuron``.
     '''
-    return np.array(((old_offs[0], old_offs[1] + spacing[1]),
-                     (new_offs[0], old_offs[1] + spacing[1]),
-                     (new_offs[0], old_offs[1] + spacing[1] - diameter),
-                     (old_offs[0], old_offs[1] + spacing[1] - diameter)))
-
-
-def _spacingx(node, max_dims, xoffset, xspace):
-    '''Determine the spacing of the current node depending on the number
-       of the leaves of the tree
+    if isinstance(neuron, Neurite):
+        return Dendrogram(neuron.root_node)
+    SomaSection = namedtuple('NeuromSection', ['id', 'type', 'children', 'points'])
+    soma_section = SomaSection(
```

Comment:
is there a way to do this with having a 'fake' soma section?  I'm thinking that `Dendrogram()` could check the type, and then create the 'fake' data inline, instead of it being bundled into an object here, and basically decomposed in the `__init__`